### PR TITLE
[servicenow] Add product version and remove `logs` from configuration page description.

### PR DESCRIPTION
**Pull request by muskan-agarwal26**
Type of change
- Enhancement

## Proposed commit message

- Add support for the timestamp_field input from the user to store inside cursor for data retrieval after intervals and also map with the `@timestamp` field.
- Provide support for ingesting Parquet files through AWS S3 or SQS.
- Add product version in the Readme.
- Remove `logs` keyword from the description on configuration page.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## How to test this PR locally

- Clone integrations repo.
- Install the elastic package locally.
- Start the elastic stack using the elastic package.
- Move to integrations/packages/servicenow directory.
- Run the following command to run tests.

> elastic-package test
```
elastic-package test
2024/11/20 15:23:50  INFO New version is available - v0.107.1. Download from: https://github.com/elastic/elastic-package/releases/tag/v0.107.1
Run asset tests for the package
2024/11/20 15:23:50  INFO License text found in "/root/github/integrations/LICENSE.txt" will be included in package
--- Test results for package: servicenow - START ---
╭────────────┬─────────────┬───────────┬─────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME                                                           │ RESULT │ TIME ELAPSED │
├────────────┼─────────────┼───────────┼─────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ servicenow │             │ asset     │ dashboard servicenow-014c13b4-59bc-4bfa-bcb2-fdc1828020bf is loaded │ PASS   │       1.22µs │
│ servicenow │             │ asset     │ dashboard servicenow-0305f883-12a8-4021-af73-fbe05e797c3f is loaded │ PASS   │        232ns │
│ servicenow │             │ asset     │ dashboard servicenow-22d413b7-ffb5-450f-b5f8-7dcd54fab22f is loaded │ PASS   │        227ns │
│ servicenow │             │ asset     │ dashboard servicenow-297414cc-3b65-40b1-b15b-e33cf676b3bf is loaded │ PASS   │        222ns │
│ servicenow │             │ asset     │ dashboard servicenow-488772bc-4ce5-48c6-9e8a-d948439dbf39 is loaded │ PASS   │        237ns │
│ servicenow │             │ asset     │ dashboard servicenow-648308f9-ebdf-46fe-b6ce-b90dfcf2d1dd is loaded │ PASS   │        260ns │
│ servicenow │             │ asset     │ dashboard servicenow-b182c6a6-3b4e-413f-899d-dd255b1a3623 is loaded │ PASS   │        297ns │
│ servicenow │             │ asset     │ dashboard servicenow-b4795d3e-8c8b-48b2-900e-edc89db4d336 is loaded │ PASS   │        301ns │
│ servicenow │             │ asset     │ dashboard servicenow-d2eb3f0c-4ddb-4683-afab-2820d2616ca2 is loaded │ PASS   │        312ns │
│ servicenow │             │ asset     │ dashboard servicenow-d8977587-d9b4-4487-b321-a8402da07d48 is loaded │ PASS   │        328ns │
│ servicenow │             │ asset     │ dashboard servicenow-d929c3a1-7af8-428e-8357-8b2e8d3e3050 is loaded │ PASS   │        347ns │
│ servicenow │             │ asset     │ dashboard servicenow-e6093b0d-38a6-4b05-8354-4bbaef28b667 is loaded │ PASS   │        363ns │
│ servicenow │             │ asset     │ search servicenow-00004bce-eac2-46d4-bb4e-ce6849c50a36 is loaded    │ PASS   │        249ns │
│ servicenow │             │ asset     │ search servicenow-01396f16-041b-4ec5-b104-d29fe68f09ad is loaded    │ PASS   │        276ns │
│ servicenow │             │ asset     │ search servicenow-046f04ce-7f31-4929-81fa-34561fb369f0 is loaded    │ PASS   │        274ns │
│ servicenow │             │ asset     │ search servicenow-1c323098-190c-4592-aca5-dd9e64de540b is loaded    │ PASS   │        306ns │
│ servicenow │             │ asset     │ search servicenow-26f2b1af-7d7d-4578-b325-322ac4c783fa is loaded    │ PASS   │        324ns │
│ servicenow │             │ asset     │ search servicenow-44db573d-f3e6-4f59-b1a4-328040220991 is loaded    │ PASS   │        311ns │
│ servicenow │             │ asset     │ search servicenow-722e7751-818d-41e2-b7ac-fcc90bb8241b is loaded    │ PASS   │        305ns │
│ servicenow │             │ asset     │ search servicenow-a5b0060e-4ace-4c9f-96e1-2d815bc9dd28 is loaded    │ PASS   │        334ns │
│ servicenow │             │ asset     │ search servicenow-afa470f8-4e19-4fee-8b17-5649a7b0ae51 is loaded    │ PASS   │        343ns │
│ servicenow │             │ asset     │ search servicenow-b9cc24b4-0cfe-458c-8645-23838623ca09 is loaded    │ PASS   │        356ns │
│ servicenow │             │ asset     │ search servicenow-e28e5e66-d00d-4e2c-af2e-a09d0d261b05 is loaded    │ PASS   │        422ns │
│ servicenow │ event       │ asset     │ index_template logs-servicenow.event is loaded                      │ PASS   │        356ns │
│ servicenow │ event       │ asset     │ ingest_pipeline logs-servicenow.event-0.1.1 is loaded               │ PASS   │        182ns │
╰────────────┴─────────────┴───────────┴─────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: servicenow - END   ---
Done
Run pipeline tests for the package
--- Test results for package: servicenow - START ---
╭────────────┬─────────────┬───────────┬───────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME                                 │ RESULT │ TIME ELAPSED │
├────────────┼─────────────┼───────────┼───────────────────────────────────────────┼────────┼──────────────┤
│ servicenow │ event       │ pipeline  │ (ingest pipeline warnings test-event.log) │ PASS   │ 524.113373ms │
│ servicenow │ event       │ pipeline  │ test-event.log                            │ PASS   │ 2.781682512s │
╰────────────┴─────────────┴───────────┴───────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: servicenow - END   ---
Done
Run policy tests for the package
--- Test results for package: servicenow - START ---
No test results
--- Test results for package: servicenow - END   ---
Done
Run static tests for the package
--- Test results for package: servicenow - START ---
╭────────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├────────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ servicenow │ event       │ static    │ Verify sample_event.json │ PASS   │  270.99497ms │
╰────────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: servicenow - END   ---
Done
Run system tests for the package
2024/11/20 15:24:17  INFO License text found in "/root/github/integrations/LICENSE.txt" will be included in package
2024/11/20 15:25:24  INFO Write container logs to file: /root/github/integrations/build/container-logs/servicenow-1732096524308361349.log
--- Test results for package: servicenow - START ---
╭────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ servicenow │ event       │ system    │ default   │ PASS   │ 47.325079073s │
╰────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: servicenow - END   ---
Done
```

### CODE REVIEW

File: packages/servicenow/data_stream/event/manifest.yml [hunk 1]
```diff
@@ -59,6 +59,14 @@ streams:
         multi: false
         required: true
         show_user: true
+      - name: timestamp_field
+        type: text
+        title: Timestamp Field
+        multi: false
+        required: true
+        show_user: true
+        description: Timestamp to store inside cursor to retrieve tables after intervals.
```
Comment (kcreddy) [inline packages/servicenow/data_stream/event/manifest.yml:68]:
```suggestion
        description: Name of the timestamp field from the table to store inside cursor to retrieve tables after intervals.
```

File: packages/servicenow/data_stream/event/manifest.yml [hunk 2]
```diff
@@ -285,6 +286,83 @@ streams:
...
+        multi: false
+        required: true
+        show_user: true
+      - name: timestamp_field
+        type: text
+        title: Timestamp Field
+        multi: false
+        required: true
+        show_user: true
+        description: Timestamp to store inside cursor to retrieve tables after intervals.
```
Comment (kcreddy) [inline packages/servicenow/data_stream/event/manifest.yml:364]: Change the description here as well.

File: packages/servicenow/data_stream/event/manifest.yml [hunk 3]
```diff
@@ -285,6 +286,83 @@ streams:
...
+        default: false
+        description: The parquet codec is used to decode parquet compressed data streams.
+      - name: decoding_batch_size
+        type: integer
+        title: Batch Size of Parquet Decoding
+        multi: false
+        required: false
+        show_user: false
+        default: 1
+        description: The batch_size attribute can be used to specify the number of records to read from the parquet stream at a time.
```
Comment (kcreddy) [inline packages/servicenow/data_stream/event/manifest.yml:342]:
```suggestion
        description: Number of records to read from the parquet stream at a time.
```

File: packages/servicenow/data_stream/event/manifest.yml [hunk 4]
```diff
@@ -285,6 +286,83 @@ streams:
...
+      - name: decoding_process_parallel
+        required: false
+        show_user: false
+        title: Process Parellel of Parquet Decoding
+        type: bool
+        multi: false
+        default: false
+        description: If the process_parallel attribute is set to true then functions which read multiple columns will read those columns in parallel from the parquet stream with a number of readers equal to the number of columns.
```
Comment (kcreddy) [inline packages/servicenow/data_stream/event/manifest.yml:350]:
```suggestion
        description: If set to true, functions which read multiple columns will read those columns in parallel from the parquet stream with a number of readers equal to the number of columns. See [Parquet Codec](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-aws-s3.html#attrib-decoding-parquet) for details.
```

File: packages/servicenow/data_stream/event/manifest.yml [hunk 5]
```diff
@@ -285,6 +286,83 @@ streams:
...
+        description: The parquet codec is used to decode parquet compressed data streams.
+      - name: decoding_batch_size
+        type: integer
+        title: Batch Size of Parquet Decoding
+        multi: false
+        required: false
+        show_user: false
+        default: 1
+        description: The batch_size attribute can be used to specify the number of records to read from the parquet stream at a time.
+      - name: decoding_process_parallel
```
Comment (kcreddy) [inline packages/servicenow/data_stream/event/manifest.yml:343]: In the S3 input, these options are specific to parquet. Any reason not to use the naming like `decoding_parquet_process_parallel`? Similar for not naming `decoding_batch_size` to `decoding_parquet_batch_size`

https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-aws-s3.html#attrib-decoding-parquet

File: packages/servicenow/changelog.yml [hunk 6]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "0.1.1"
```
Comment (kcreddy) [inline packages/servicenow/changelog.yml:2]:
```suggestion
- version: "0.2.0"
```

File: packages/servicenow/manifest.yml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 format_version: 3.2.1
 name: servicenow
 title: "ServiceNow"
-version: 0.1.0
+version: 0.1.1
```
Comment (kcreddy) [inline packages/servicenow/manifest.yml:4]:
```suggestion
version: 0.2.0
```

File: packages/servicenow/manifest.yml [hunk 8]
```diff
@@ -71,11 +71,11 @@ policy_templates:
     description: Collect ServiceNow logs.
     inputs:
       - type: cel
-        title: Collect ServiceNow table logs via API
-        description: Collect table logs from ServiceNow instance via API.
+        title: Collect ServiceNow tables via API
+        description: Collect tables from ServiceNow instance via API.
```
Comment (kcreddy) [inline packages/servicenow/manifest.yml:75]: There is also a reference to `table logs` in Knowledge dashboard. Please change it as well.

File: packages/servicenow/data_stream/event/manifest.yml (hunk 5 above)
Comment (muskan-agarwal26) [reply packages/servicenow/data_stream/event/manifest.yml:343]: I had referenced these variables from [Amazon Security Lake](https://github.com/elastic/integrations/blob/main/packages/amazon_security_lake/data_stream/event/manifest.yml#L42C15-L42C34) integration, but I will go ahead and change the names according to the documentation.

**Review by kcreddy — APPROVED** [review]

**Pull request merged**